pkg/generators/snippets/constructor.go

```diff
@@ -21,17 +20,30 @@ func New$.type|raw$() *$.type|raw$ {
 }
 
 func GenerateConstructorForObjectMeta(t *types.Type) (string, generator.Args) {
-	args := generator.Args{
-		"type": t,
-	}
+	args := defaultGeneratorArgs(t)
 
 	raw := `// New$.type|raw$ is an autogenerated constructor.
-func New$.type|raw$(name string) *$.type|raw$ {
-	o := &$.type|raw${}
+func New$.type|raw$(name string) $.pointer$$.type|raw$ {
+	o := $.ampersand$$.type|raw${}
 	o.ObjectMeta.Name = name
 	return o
 }
 
 `
 	return raw, args
 }
+
+func defaultGeneratorArgs(t *types.Type) generator.Args {
```

Comment:
```suggestion
func defaultGeneratorArgs(t *types.Type, valueReceiver bool) generator.Args {
```

The builder will need to supply this information.

Reply:
gotcha. I used `pointerReceiver` as the argument instead as it works better with the code.